### PR TITLE
SNOW-1738538 Fix multi ast eval validation in decoding expectation test ast

### PR DESCRIPTION
**Pull request by sfc-gh-evandenberg**
Fix multi ast eval validation in decoding expectation test ast

### CODE REVIEW

File: src/snowflake/snowpark/_internal/ast_utils.py [hunk 1]
```diff
@@ -1342,7 +1342,19 @@ def ClearTempTables(message: proto.Request) -> None:
 
 def base64_str_to_request(base64_str: str) -> proto.Request:
```
Comment (sfc-gh-lspiegelberg) [inline src/snowflake/snowpark/_internal/ast_utils.py:1343]: I think we should be here a bit more careful with API naming and semantics.

Technically, `"\n"` is not a valid character in base64. The function itself right now does exactly one thing, which is taking a valid base64_str (for which the user is responsible, we can do a check) and decodes it. I would suggest to keep the basic function as is, but for convenience add a new one which does the repeated decode and then merges the result together.

I.e.,
```
def base64_str_to_request(base64_str: str) -> proto.Request:
  ....
 < add check that characters are only in `[0-9+/=a-zA-Z]` which is base64 encoding>
 
 def base64_lines_to_requst(base64_lines: str) -> proto.Request:
    # allow here character "\n"
    requests = [base64_str_to_request]
    return merge_requests(requests)
```

This is I think anyways the logic you have in mind here, but easier to understand

Comment (sfc-gh-evandenberg) [reply src/snowflake/snowpark/_internal/ast_utils.py:1343]: Updated

File: src/snowflake/snowpark/_internal/ast_utils.py [hunk 2]
```diff
@@ -1342,12 +1342,31 @@ def ClearTempTables(message: proto.Request) -> None:
...
     message = proto.Request()
-    message.ParseFromString(base64.b64decode(base64_str.strip()))
+    message.ParseFromString(base64.b64decode(base64_str))
     return message
 
 
-def base64_str_to_textproto(base64_str: str) -> str:
-    request = base64_str_to_request(base64_str)
+def merge_requests(messages: List[proto.Request]) -> proto.Request:
+    message = proto.Request()
```
Comment (sfc-gh-lspiegelberg) [inline src/snowflake/snowpark/_internal/ast_utils.py:1350]: nit: maybe add docstring comment on what the function does (i.e., merge all requests into single request. Order is the same as messages).

nit: rename potentially messages -> requests?

**Review by sfc-gh-lspiegelberg — APPROVED** [review]

File: src/snowflake/snowpark/_internal/ast_utils.py (hunk 2 above)
Comment (sfc-gh-lspiegelberg) [reply src/snowflake/snowpark/_internal/ast_utils.py:1350]: Empty case is also not handled in this function. I.e.

```
if not messages:
   return message
```

Comment (sfc-gh-evandenberg) [reply src/snowflake/snowpark/_internal/ast_utils.py]: Updated

**Pull request merged**